irc/config.go

```diff
@@ -1224,6 +1239,6 @@ func compileGuestRegexp(guestFormat string, casemapping Casemapping) (standard,
 	if err != nil {
 		return
 	}
-	folded, err = regexp.Compile(fmt.Sprintf("^%s(.*)%s$", initialFolded, finalFolded))
+	folded, err = utils.CompileGlob(fmt.Sprintf("%s*%s", initialFolded, finalFolded))
```

Comment:
The desire to casefold both sides of the `*` separately stems from PRECIS being a black box; in particular, it might reject a string mixing `*` and non-Latin chars. I'm making some of the other simplifications.